libs/vscode/project-details/src/lib/new-project-details-preview.ts

```diff
@@ -0,0 +1,560 @@
+import { PDVData } from '@nx-console/shared/types';
+import { onWorkspaceRefreshed } from '@nx-console/vscode/lsp-client';
+import { getPDVData } from '@nx-console/vscode/nx-workspace';
+import { join } from 'path';
+import {
+  commands,
+  Disposable,
+  ExtensionContext,
+  Uri,
+  ViewColumn,
+  WebviewPanel,
+  window,
+} from 'vscode';
+import {
+  assign,
+  createActor,
+  enqueueActions,
+  fromPromise,
+  setup,
+} from 'xstate';
+import { ProjectDetailsPreview } from './project-details-preview';
+import { handleGraphInteractionEventBase } from '@nx-console/vscode/graph-base';
+import { getGraphWebviewManager } from '@nx-console/vscode/project-graph';
+
+export class NewProjectDetailsPreview implements ProjectDetailsPreview {
+  private webviewPanel: WebviewPanel;
+
+  projectRoot: string | undefined;
+
+  constructor(private path: string, private context: ExtensionContext) {
+    this.webviewPanel = window.createWebviewPanel(
+      'nx-console-project-details',
+      `Project Details`,
+      ViewColumn.Beside,
+      {
+        enableScripts: true,
+      }
+    );
+    const machine = setup({
+      types: {
+        context: {} as Partial<PDVData> & {
+          multiSelectedProject: string | undefined;
+        },
+      },
+      actors: {
+        loadPDVData: fromPromise(async () => {
+          return await getPDVData(path);
+        }),
+      },
+      actions: {
+        renderLoading: () => this.renderLoading(),
```

Comment:
Not really an XState expert, but it feels like the render side-effects aren't part of the workflow. Maybe instead it should be a handler to `actor.subscribe` that renders the webview depending on the current state?

Reply:
Not an XState expert either, so you might be completely right. My thought process goes like this:
While the simple rendering cases (loading, error) could easily be done with a state subscription, the PDV cases with their child states & `reRenderPDV` logic seem more complicated. So in the subscription we would have to branch depending on the previous state of the machine: is it just entering the `showingPDV.idle` state or was it in `showingPDV.refreshing` before? 
So while we would remove the actions from the machine, we would essentially write a subscription that is tightly coupled to the machine implementation, which seems like it's just spreading out the logic instead of actually decoupling it.